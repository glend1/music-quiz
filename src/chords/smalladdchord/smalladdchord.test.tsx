import { render, screen } from "@testing-library/react";
import { useState } from "react";
import { TChords } from "../types";
import sharedStyles from "../../../styles/shared.module.css";
import userEvent from "@testing-library/user-event";
import SmallAddChord from "./smalladdchord";

function MockAddChord() {
	const [chords, setChords] = useState<TChords>({ test: [] });
	const [active, setActive] = useState("test");
	return (
		<>
			<h3>{active}</h3>
			<SmallAddChord setChords={setChords} setActive={setActive} />
		</>
	);
}

describe("addchord", () => {
	it("Should render the element", () => {
		let component = render(<MockAddChord />);
		expect(component.baseElement.querySelectorAll(".add")[0]).toBeVisible();
	});
	it("Should add a chord", async () => {
		const component = render(<MockAddChord />);
		await userEvent.click(component.baseElement.querySelectorAll(".add")[0]);
		expect(screen.getByRole("heading")).not.toHaveTextContent("test");
	});
});
