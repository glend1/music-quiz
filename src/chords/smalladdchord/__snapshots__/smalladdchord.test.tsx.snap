// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addchord Should add a chord 1`] = `"<div><div class="container"><div class="add clickable"></div></div></div>"`;
